core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1007,6 +1007,49 @@ public Builder setBranchSnapshot(long snapshotId, String branch) {
       return this;
     }
 
+    public Builder setTag(String tag, SnapshotRef ref) {
+      SnapshotRef existingRef = refs.get(tag);
+      if (existingRef != null && existingRef.equals(ref)) {
+        return this;
+      }
+      long snapshotId = ref.snapshotId();
```

Comment:
Please remember to format code! There are several places that are missing require whitespace.

Reply:
Ah my bad, I will make sure on the next iteration to cleanup any white space/newline I missed